### DISCUSSION TITLE
Stupid Apple HK limit broke everything..

### DISCUSSION THREAD

**Discussion by sahafeez** [kuestess/homebridge-platform-insteonlocal#229] [General]
Everything was working great. As part of the move to this I made all the device names make sense. So for example - the top plug next to the bar is named:

T@Bar

I rebooted the main HomeKit hub, an AppleTV. Another one took over. Nothing worked. All devices where marked as unresponsive. Did all the normal checking, rebooting, troubleshooting, etc. wiping cache, etc. Damn it, nothing worked. Deleted bridge, fun, re-sort 104 items in Home. I noticed one was misnamed so I tried to fix it by adding the location name. In this case Kitchen @Dining. Apple said, "nope!" you cannot use that name. Ok lets try Kitchen -Dining. Took it. Except the icon showed just " -Dining". WTF?

Okay vi config.json :109,600s/ @/./ etc..

What I think happened is it worked on the first import but the handoff to the hub broke it. This is a guess.

Anyway...fun.

@kuestess might want to figure out valid Apple item names and put in check for what people call stuff or just document it. How the hell in 2022 can I NOT sort for @()'&$&% in string! 

Sending bitchy email to friends at Apple :)

**Comment by kuestess** [comment]
Sorry, I'm not laughing at your expense :-).  Hey, at least you figured it out - stupid crap like that can be frustrating.  Wonder if someone has a regex built - that would be a fairly easy thing to check.  Could automatically remove the illegal characters.

**Reply by sahafeez** [reply]
@kuestess Ha! All the issues I am opening are due to my own OCD on getting things perfect. All good! :) Pissed that I found a naming system that makes sense and I cannot use it.

**Comment by rwgunther** [comment]
@sahafeez Not sure about the special characters in names, but I suspect I know what’s going on with the -Dining name thing. If a name of a HomeKit accessory or service starts with the name of the room it’s in, the Home app will suppress the name of that room in the title. It’s probably so you don’t end up with 10 Dining Room Blah Blah tiles, with the name of the actual thing getting cut off. If you include that accessory in your favorites so it appears on Home’s Home screen [don’t even get me started about that], it will probably appear with the full name.

**Reply by sahafeez** [reply]
That makes sense. I hardly every use the items in the rooms, just Home Screen so I did not notice.
